tests/translator/output/function_with_custom_codedeploy_deployment_preference.json

```diff
@@ -1,142 +1,864 @@
+          "Events": [
+            "DEPLOYMENT_FAILURE", 
+            "DEPLOYMENT_STOP_ON_ALARM", 
+            "DEPLOYMENT_STOP_ON_REQUEST"
+          ]
+        }, 
+        "ServiceRoleArn": {
+          "Fn::GetAtt": [
+            "CodeDeployServiceRole", 
+            "Arn"
+          ]
+        }, 
+        "DeploymentConfigName": {
+          "Fn::Sub": "CustomDeployment"
+        }, 
+        "DeploymentStyle": {
+          "DeploymentType": "BLUE_GREEN", 
+          "DeploymentOption": "WITH_TRAFFIC_CONTROL"
+        }
+      }
+    }, 
+    "CustomWithCondition2Version640128d35d": {
+      "DeletionPolicy": "Retain", 
+      "Type": "AWS::Lambda::Version", 
+      "Properties": {
+        "FunctionName": {
+          "Ref": "CustomWithCondition2"
+        }
+      }
+    }, 
+    "CustomWithCondition": {
+      "Type": "AWS::Lambda::Function", 
+      "Properties": {
+        "Handler": "hello.handler", 
+        "Code": {
+          "S3Bucket": "sam-demo-bucket", 
+          "S3Key": "hello.zip"
+        }, 
+        "Role": {
+          "Fn::GetAtt": [
+            "CustomWithConditionRole", 
+            "Arn"
+          ]
+        }, 
+        "Runtime": "python2.7", 
+        "Tags": [
+          {
+            "Value": "SAM", 
+            "Key": "lambda:createdBy"
+          }
+        ]
+      }
+    }, 
+    "CustomWithSub": {
+      "Type": "AWS::Lambda::Function", 
+      "Properties": {
+        "Handler": "hello.handler", 
+        "Code": {
+          "S3Bucket": "sam-demo-bucket", 
+          "S3Key": "hello.zip"
+        }, 
+        "Role": {
+          "Fn::GetAtt": [
+            "CustomWithSubRole", 
+            "Arn"
+          ]
+        }, 
+        "Runtime": "python2.7", 
+        "Tags": [
+          {
+            "Value": "SAM", 
+            "Key": "lambda:createdBy"
+          }
+        ]
+      }
+    }, 
+    "ServerlessDeploymentApplication": {
+      "Type": "AWS::CodeDeploy::Application", 
+      "Properties": {
+        "ComputePlatform": "Lambda"
+      }
+    }, 
+    "NormalWithRefVersion640128d35d": {
+      "DeletionPolicy": "Retain", 
+      "Type": "AWS::Lambda::Version", 
+      "Properties": {
+        "FunctionName": {
+          "Ref": "NormalWithRef"
+        }
+      }
+    }, 
+    "NormalWithSubDeploymentGroup": {
+      "Type": "AWS::CodeDeploy::DeploymentGroup", 
+      "Properties": {
+        "ApplicationName": {
+          "Ref": "ServerlessDeploymentApplication"
+        }, 
+        "AutoRollbackConfiguration": {
+          "Enabled": true, 
+          "Events": [
+            "DEPLOYMENT_FAILURE", 
+            "DEPLOYMENT_STOP_ON_ALARM", 
+            "DEPLOYMENT_STOP_ON_REQUEST"
+          ]
+        }, 
+        "ServiceRoleArn": {
+          "Fn::GetAtt": [
+            "CodeDeployServiceRole", 
+            "Arn"
+          ]
+        }, 
+        "DeploymentConfigName": {
+          "Fn::Sub": [
+            "CodeDeployDefault.Lambda${ConfigName}", 
```

Comment:
Here's an example of the "Sub nested in Sub" case that the code explicitly fixes.